src/services/userToRoomJoiner.js

```diff
@@ -0,0 +1,61 @@
+const { User, Room } = require('../entities')
+
+class UserToRoomJoiner {
+    constructor({ userRepo, roomRepo }) {
+        this.userRepo = userRepo
```

Comment:
Where do `repos` come from here?
I mean they are not required, although they and their methods are still used here.

Reply:
It is passed in constructor `UserToRoomJoiner`. The main rule is business logic should not depends on technical details (such knex, pg)

Reply:
So `{ userRepo, roomRepo }` is a destructuring of whatever is being passed into constructor, right?

Also, sorry but I get confused by the naming.......in the line `this.userRepo = userRepo`............
Is the blue 'userRepo' created by the constructor property?

I think it's not, I suppose it's not connected, meaning that if we change `constructor({ userRepo, roomRepo })` to `constructor({ lalala, roomRepo })`, we wouldn't need to change the blue 'userRepo', only the white one after the '='.

This may sound silly, but as I said I get confused when stuff has the same name. Because that makes me think "is this necessary?" / "is there some deeper sense in them being titled the same?"

Reply:
```
So { userRepo, roomRepo } is a destructuring of whatever is being passed into constructor, right?
```
  Right

```
Also, sorry but I get confused by the naming.......in the line this.userRepo = userRepo............
Is the blue 'userRepo' created by the constructor property?
```

you can understand `this` as  a simple object. It is an empty one by default. You just assign a property to `this`. The key point of `this` it is accessible from other instance methods.